src/Grpc.Net.Client/Internal/GrpcProtocolConstants.cs

```diff
@@ -103,15 +115,29 @@ static GrpcProtocolConstants()
             // but in case it isn't then don't include version in user-agent.
             if (assemblyVersion != null)
             {
-                userAgent += "/" + assemblyVersion.Version;
+                // 1.24.0
+                userAgent += $"/{assemblyVersion.Version} ";
             }
 
-            UserAgentHeader = "User-Agent";
-            UserAgentHeaderValue = userAgent;
-            TEHeader = "TE";
-            TEHeaderValue = "trailers";
+            // (.NET 5.0.7;
+            userAgent += $"({RuntimeInformation.FrameworkDescription}; ";
```

Comment:
nit: RuntimeInformation.FrameworkDescription is available starting from .NET Framework 4.7.1+, but we had plans to support the legacy net46+ (for grpc-web and grpc over winhttp2 handler). 
So if we added the net461 target, this won't build.

What you currently have is fine but perhaps worth adding a TODO that if net46x targets are added in the future, this will need more work (we can get inspired with what Grpc.Core does here. https://github.com/grpc/grpc/pull/25889/files#diff-e0b4c69507f528047f6f1d956277df9468e730ca55066821869e7b140d96b66dR207